### ISSUE TITLE
Feature Request: Support for InfluxDB 2

### ISSUE THREAD

**Issue by ShakataGaNai** [grafana/grafana#17561]
**What would you like to be added**:
The first [Alpha of InfluxDB 2.0](https://www.influxdata.com/blog/influxdb-2-0-alpha-release-and-the-road-ahead/) has been released. It's quite a bit different in terms of access and data organization than the 1.x series. So support for these new feature sets would be appreciated.

**Why is this needed**:
InfluxDB 2 is a fairly major update for that datasource with a lot of new features. It stands to reason that people will want to upgrade to it.

**Comment by bergquist** [comment]
related https://grafana.com/plugins/grafana-influxdb-flux-datasource

More support to come.

**Comment by Sarke** [comment]
Do not use that plugin with 2.0, as it is not supported.  CPU will spike and crash the InfluxDB server.

**Comment by lafrech** [comment]
The CHANGELOG reads:

```
[5.3.0] - 2019-06-11

    Update packages
    Add circleci publishing
    Add support for Influx V2.0 Alpha Server
```

But the README says:

```
Make sure you have Grafana and Influx v1.7.4 or above running. (Note: v2.x.x is not yet supported.)
```

I haven't tried it yet so I can't tell which is true.

However, I've been watching @jtlisi's [InfluxDays 2019 NYC talk](https://www.youtube.com/watch?v=cD2yOzQWe8Q) and the interface provided by this plugin looks very raw. It exposes a window to write flux queries into, rather than dropdown menus to build queries.

Is there an effort, in this plugin or in parallel work, to build an interface UI like the existing one (and like Chronograf) where the user builds his queries with only mouse clicks?

While the raw Flux interface might be much more powerful, we find interest in the query assistant.

The talk is one year old, so things may have changed in the meantime, but my understanding is that the raw Flux query box was a design choice, not a current limitation.

**Comment by memetb** [comment]
One of the big regressions is that the flux data source doesn't support alerts. I can't tell if this is a grafana side (i.e in the plugin) or server side (influxdb flux language itself) problem.

**Comment by russorat** [comment]
Support for InfluxDB 2 has been added to Grafana. I think this issue can be closed.

https://v2.docs.influxdata.com/v2.0/visualize-data/other-tools/grafana/

**Comment by lafrech** [comment]
AFAIK, Grafana supports

- InfluxQL (InfluxDB 1) with a nice query-builder GUI suitable for people with no InfluxQL knownledge (with a raw query mode for more complex needs)
- Flux (InfluxDB 2) with a textbox GUI allowing a more advanced user to enter raw Flux queries

Grafana doesn't have a GUI tool to build Flux queries.

Since InfluxQL finally made it back into InfluxDB 2, we can say InfluxDB 2 query building is supported by the legacy InfluxQL mode of the InfluxDB data source.

I don't know if there are plans to drop InfluxQL at some point in InfluxDB in which case a Flux query-builder would be an important feature request again.

Query-builder GUI is a critical feature to us. I'm using Grafana to expose timeseries datasets to engineers who wouldn't build their own queries but are quite comfortable with the Grafana GUI. Using Grafana has been a great improvement to our workflow, so thanks for that.

**Comment by gabor** [comment]
@lafrech i'm glad grafana is useful in your workflow!

**Comment by gabor** [comment]
@ShakataGaNai grafana supports influxdb2. the support for the influxql query language is stronger, the support for the flux query language is currently in beta. so i consider this issue resolved. if there is a specific feature you need and is not supported, please open a separate issue. thanks!
